app/controllers/api/reference/regions_controller.rb

```diff
@@ -4,7 +4,13 @@ module Api
   module Reference
     class RegionsController < ApiController
       def index
-        render json: Region.all.includes(:locations)
+        render json: Region.all.includes(locations: :suppliers), include: included_relationships
```

Comment:
Annoyingly the `LocationSerializer` includes associated suppliers, so `suppliers` are eager fetched here to avoid N+1 query for every location.

Reply:
Can't we use `includes` only when we have the `location.supplier`include in the query param? I don't think the location serializer includes suppliers by default. if you remove suppliers in this line:

https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/master/spec/serializers/location_serializer_spec.rb#L11

The included test fails because it was expecting a supplier.

Reply:
This does raise an interesting/potentially valuable optimisation question - we should match the includes in index actions for the database query to make sure we're not doing N+1 everywhere.

I think we can achieve this in a general way to reduce N+1 in the future.

Reply:
The `LocationSerializer` is somewhat strange and does cause lots of N+1 queries unless the suppliers are prefetched. The suppliers don't get included in the returned JSON payload, but are listed as relationships of the included locations - which due to the magic of AMS means running a DB query for each location :(